src/include/miopen/groupnorm/problem_description.hpp

```diff
@@ -0,0 +1,162 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <miopen/problem_description_base.hpp>
+#include <miopen/activ.hpp>
+#include <miopen/tensor.hpp>
+
+#include <cassert>
+#include <string>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace groupnorm {
+
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(miopenNormMode_t mode_,
+                       const TensorDescriptor& xDesc_,
+                       const TensorDescriptor& weightDesc_,
+                       const TensorDescriptor& biasDesc_,
+                       const TensorDescriptor& yDesc_,
+                       const TensorDescriptor& meanDesc_,
+                       const TensorDescriptor& rstdDesc_,
+                       uint64_t num_groups_,
+                       float epsilon_)
+        : mode(mode_),
+          xDesc(xDesc_),
+          weightDesc(weightDesc_),
+          biasDesc(biasDesc_),
+          yDesc(yDesc_),
+          meanDesc(meanDesc_),
+          rstdDesc(rstdDesc_),
+          num_groups(num_groups_),
+          epsilon(epsilon_)
+    {
+        if(xDesc.GetType() != yDesc.GetType())
+        {
+            MIOPEN_THROW(miopenStatusBadParm, "GroupNormForward: Tensor types do not match.");
+        }
+        if(meanDesc.GetType() != rstdDesc.GetType())
+        {
+            MIOPEN_THROW(miopenStatusBadParm, "GroupNormForward: Tensor types do not match.");
+        }
+        if(xDesc.GetLengths() != yDesc.GetLengths())
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "GroupNormForward: Tensor dimension lengths do not match.");
+        }
+        if((num_groups < 1) || (xDesc.GetLengths().size() < 3) ||
+           (xDesc.GetLengths()[1] % num_groups != 0))
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "GroupNormForward: The channel size of input tensor should be divisible "
+                         "by num_groups.");
+        }
+        if(!(xDesc.IsPacked() && weightDesc.IsPacked() && biasDesc.IsPacked() && yDesc.IsPacked() &&
+             meanDesc.IsPacked() && rstdDesc.IsPacked()))
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "GroupNormForward: Unpacked tensors are not supported.");
```

Comment:
This kind of problems seems valid and thus should be constructible. If we do not have any solution, then the library will MIOPEN_THROW somewhere else.

Reply:
@kyeonghwanryu 
> 1. remove MIOPEN_THROW when _not all tensors_ are ~not~ packed in the constructor

Yes, because the problem is valid.

[Notice] "_all tensors are not packed_" == "_all tensors are unpacked_" != "_not all tensors are packed_" ;)

> 2. instead, return false in GroupNormForward::IsApplicable with same condition

Yes. If GroupNormForward supports only packed tensors, then if must return false in this case.

Reply:
Fixed as your recommendation.

Reply:
Sorry for incorrect usage of a negative sentence :)